### PR TITLE
Schedule filter

### PR DESCRIPTION
**Pull request by DiamondAhuja**
right now by default there are no schedules because it takes it as current date, and there are no schedules in the db for the current date
https://github.com/LowerEarthOrbiters/Lower_Earth_Orbiters/assets/129548993/8fdb3d3d-57ad-40a8-862b-a4a357053fcb

### CODE REVIEW

**Comment by quinnha** [comment]
image

some changes
- show all the overpasses by default (its currently empty before you apply filter)
- show the date range by default (instead of having to click filter button to show, use filter button to execute filter)

looks clean otherwise

File: src/leo-server-app/src/routes/schedule.ts [hunk 1]
```diff
@@ -388,21 +389,36 @@ router.get(
...
 
-    const convertedTime = new Date(time);
+    if (!mongoose.isValidObjectId(satelliteId)) {
+      return res.status(500).json({ error: "Invalid ID" });
+    }
 
+    const convertedStartTime = new Date(startTime);
+    convertedStartTime.setHours(convertedStartTime.getHours() + 5);
+    const convertedEndTime = new Date(endTime);
+    convertedEndTime.setHours(convertedEndTime.getHours() + 29);
```
Comment (Umang-Rajkarnikar) [inline src/leo-server-app/src/routes/schedule.ts:410]: What is the +5 and +14 doing?

Comment (DiamondAhuja) [reply src/leo-server-app/src/routes/schedule.ts]: changed back to original but mongo needs to be updated

**Comment by DiamondAhuja** [comment]
added dropdown for filter + resolved quinn's comments
https://github.com/LowerEarthOrbiters/Lower_Earth_Orbiters/assets/129548993/f38d25a3-81cd-428a-a63e-e030ab69411d

**Comment by quinnha** [comment]
lgtm

**Review by quinnha — APPROVED** [review]

**Review by Umang-Rajkarnikar — APPROVED** [review]
lgtm

**Pull request merged**